packages/fuel-indexer/src/database.rs

```diff
@@ -64,29 +64,28 @@ impl Database {
         let conn = self.pool.acquire().await?;
         self.stashed = Some(conn);
         debug!("Connection stashed as: {:?}", self.stashed);
-        let conn = self.stashed.as_mut().expect(
-            "No stashed connection for start transaction. Was a transaction started?",
-        );
+        let conn = self.stashed.as_mut().ok_or(crate::IndexerError::Unknown(
```

Comment:
Since we're returning an error due to the absence of a stashed transaction, can we add a variant to `IndexerError` (e.g. `NoStashedConnection`) and use it instead of the `Unknown` variant in each of the places where we attempt to retrieve a stashed connection?